File: part_7-natural-language-processing/natural_language_processing.R
# Importing the dataset
dataset = read.delim('../data_files/Restaurant_Reviews.tsv', quote='', stringsAsFactors=FALSE)

# Cleaning the texts
library(tm)
library(SnowballC)
corpus = VCorpus(VectorSource(dataset$Review))  # Making corpus
corpus = tm_map(corpus, content_transformer(tolower))   # Making Lower-case
corpus = tm_map(corpus, removeNumbers)  # Removing numbers
corpus = tm_map(corpus, removePunctuation)  # Removing Punctuation
corpus = tm_map(corpus, removeWords, stopwords())  # Removing stopwords (i.e. this, that, are etc.)
corpus = tm_map(corpus, stemDocument)   # Stemming words
corpus = tm_map(corpus, stripWhitespace)   # Stripping white-spaces

File: part_7-natural-language-processing/natural_language_processing.py
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('../data_files/Restaurant_Reviews.tsv', delimiter='\t', quoting=3)

# Cleaning the texts
import re
import nltk
# nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer

corpus = []
for i in range(0, len(dataset['Review'])):
    review = re.sub('[^a-zA-Z]', ' ', dataset['Review'][i])
    review = review.lower()
    review = review.split()
    ps = PorterStemmer()
    stopword_set = set(stopwords.words('english'))
    review = [ps.stem(word) for word in review if not word in stopword_set]
    review = ' '.join(review)
    corpus.append(review)

# Creating the Bag of Words model
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(max_features=1500)
X = cv.fit_transform(corpus).toarray()
y = dataset.iloc[:, 1].values
